network/wsPeer.go

```diff
@@ -486,6 +490,22 @@ func (wp *wsPeer) readLoop() {
 			// network maintenance message handled immediately instead of handing off to general handlers
 			wp.handleFilterMessage(msg)
 			continue
+		// the remaining valid tags: no special handling here
+		case protocol.AgreementVoteTag:
```

Comment:
I worry about these case statements getting out of sync. How about a `func IsValidTag(Tag) bool` that lives in `protocol/tags.go` so that it at least is all in one source to maintain there.